examples/simple_edit.py

```diff
@@ -152,6 +175,113 @@ async def simple_edit(
 
+        print(f"Finished writing output {output_flow_id}")
+
+
+async def interval_edit(
+    tams_url: str,
+    credentials: Credentials,
+    input_1_flow_id: UUID,
+    input_1_timerange: TimeRange,
+    input_2_flow_id: UUID,
+    input_2_timerange: TimeRange,
+    output_flow_id: UUID,
+    output_source_id: UUID,
+    cut_interval_sec: float
+) -> None:
+    """Cut between inputs 1 and 2 at fixed interval. Note: this will only handle a single page of segments"""
+    cut_interval_ts = Timestamp.from_millisec(int(cut_interval_sec * 1000))
+
+    edit_rate = Fraction(FLOW_FRAME_RATE, 1)
+
+    custom_tags = {
+        "_copy_edit_mode": "interval",
+        "_copy_edit_interval": cut_interval_ts.to_sec_nsec()
+    }
+
+    async with aiohttp.ClientSession() as session:
+        # Create output Flow
+        await put_flow(session, credentials, tams_url, output_flow_id, output_source_id, custom_tags=custom_tags)
+
+        flow_1_segments = deque(await get_segments(session, credentials, tams_url, input_1_flow_id, input_1_timerange))
+        flow_2_segments = deque(await get_segments(session, credentials, tams_url, input_2_flow_id, input_2_timerange))
+        working_time = Timestamp.from_str("0:0")
+
+        next_switch_at = working_time + cut_interval_ts
+
+        current_seg = {
+            "id": input_1_flow_id,
+            "list": flow_1_segments,
+            "offset": TimeRange.from_str(flow_1_segments[0]["timerange"]).start
+        }
+        other_seg = {
+            "id": input_2_flow_id,
+            "list": flow_2_segments,
+            "offset": TimeRange.from_str(flow_2_segments[0]["timerange"]).start
+        }
+
+        while (len(flow_1_segments) > 0 and len(flow_2_segments) > 0):
+            position_in_flow_timeline = working_time + current_seg["offset"]
+
+            # Draw a segment from the current list (and drop segments if we've passed them)
+            next_seg = current_seg["list"][0]
+            next_seg_tr = TimeRange.from_str(next_seg["timerange"]).normalise(edit_rate.numerator,
+                                                                              edit_rate.denominator)
+            if next_seg_tr.end <= position_in_flow_timeline:
+                print(f"Segment {next_seg_tr} is before current position {position_in_flow_timeline} - dropping")
+                current_seg["list"].popleft()
+                continue
+
+            next_seg_offset = Timestamp.from_str(next_seg.get("ts_offset", "0:0"))
+
+            # Work out how many seconds into the underlying segment we start
+            seg_start_offset = (working_time + current_seg["offset"]) - next_seg_tr.start
+
+            # Work out how much of the segment to include in the output
+            segment_length_remaining = next_seg_tr.end - position_in_flow_timeline
+            remaining_time_in_cut = next_switch_at - working_time
+
+            if (remaining_time_in_cut < segment_length_remaining):
```

Comment:
There will be some anomalies when doing this and other comparisons if the media rates are not integers (because of rounding errors in some of the calculations using nanoseconds), but I think on balance it isn't helpful to have those complications in this example script.

Reply:
I agree about complicating the example, but I have called this out in a comment now